lib/types.ts

```diff
@@ -128,6 +128,16 @@ export const HydratedCoinSchema = CoinSchema.extend({
 
 export type HydratedCoin = z.infer<typeof HydratedCoinSchema>
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isHydratedCoinOrEth(coin: any): coin is HydratedCoin | 'eth' {
+  return coin === 'eth' || !isRequiredString(coin)
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isEthAddressOrEth(coin: any): coin is EthAddress | 'eth' {
+  return coin === 'eth' || isRequiredString(coin)
+}
```

Comment:
I'm afraid that's not right. You cannot just assume because it's not a string; it means it's a hydratedCoin. Use zod to validate i.e. 

```
export function isEthAddressOrEth(coin: any): coin is EthAddress | 'eth' {
try {
if (coin === 'eth') return true
const x = HydratedCoinSchema.parse(coin)
return !isNull(x)
} catch(e) {
  return false
 }
```